drv/psc-seq-server/src/main.rs

```diff
@@ -187,6 +337,25 @@ const PSU_PWR_OK_NOTIF: [u32; PSU_COUNT] = [
     notifications::PSU_PWR_OK_6_MASK,
 ];
 
+/// In order to get the PMBus devices by PSU index, we need a little lookup table guy.
+const PSU_PMBUS_DEVS: [fn(TaskId) -> (I2cDevice, u8); PSU_COUNT] = [
+    i2c_config::pmbus::v54_psu0 as fn(TaskId) -> (I2cDevice, u8),
+    i2c_config::pmbus::v54_psu1 as fn(TaskId) -> (I2cDevice, u8),
+    i2c_config::pmbus::v54_psu2 as fn(TaskId) -> (I2cDevice, u8),
+    i2c_config::pmbus::v54_psu3 as fn(TaskId) -> (I2cDevice, u8),
+    i2c_config::pmbus::v54_psu4 as fn(TaskId) -> (I2cDevice, u8),
+    i2c_config::pmbus::v54_psu5 as fn(TaskId) -> (I2cDevice, u8),
```

Comment:
Casts aren't necessary because it's specified in the array type:
```suggestion
    i2c_config::pmbus::v54_psu0,
    i2c_config::pmbus::v54_psu1,
    i2c_config::pmbus::v54_psu2,
    i2c_config::pmbus::v54_psu3,
    i2c_config::pmbus::v54_psu4,
    i2c_config::pmbus::v54_psu5,
```

Reply:
I thought it didn't compile without the casts, but it looks like it does, will remove them!